pig/src/main/java/com/netflix/iceberg/pig/IcebergStorage.java

```diff
@@ -0,0 +1,313 @@
+
+    Schema schema = load(location, job).schema();
+    storeInUDFContext(ICEBERG_SCHEMA, schema);
+
+    return SchemaUtil.convert(schema);
+  }
+
+
+  @Override
+  public ResourceStatistics getStatistics(String location, Job job) {
+    LOG.info(format("[%s]: getStatistics() -> : %s", signature, location));
+
+    return null;
+  }
+
+  @Override
+  public String[] getPartitionKeys(String location, Job job) {
+    LOG.info(format("[%s]: getPartitionKeys()", signature));
+    return new String[0];
+  }
+
+  @Override
+  public void setPartitionFilter(Expression partitionFilter) {
+    LOG.info(format("[%s]: setPartitionFilter() ->  %s", signature, partitionFilter));
+  }
+
+  @Override
+  public List<String> getPredicateFields(String location, Job job) throws IOException {
+    LOG.info(format("[%s]: getPredicateFields() -> %s", signature, location));
+    Schema schema = load(location, job).schema();
+
+    List<String> result = Lists.newArrayList();
+
+    for (Types.NestedField nf : schema.columns()) {
+      switch (nf.type().typeId()) {
+        case MAP:
+        case LIST:
+        case STRUCT:
+          continue;
+        default:
+          result.add(nf.name());
+      }
+    }
+
+    return result;
+  }
+
+  @Override
+  public List<Expression.OpType> getSupportedExpressionTypes() {
+    LOG.info(format("[%s]: getSupportedExpressionTypes()", signature));
+    return asList(OP_AND, OP_OR, OP_EQ, OP_NE, OP_NOT, OP_GE, OP_GT, OP_LE, OP_LT, OP_BETWEEN, OP_IN, OP_NULL);
+  }
+
+  @Override
+  public void setPushdownPredicate(Expression predicate) throws IOException {
+    LOG.info(format("[%s]: setPushdownPredicate()", signature));
+    LOG.info(format("[%s]: Pig predicate expression: %s", signature, predicate));
+
+    com.netflix.iceberg.expressions.Expression icebergExpression = convert(predicate);
+
+    LOG.info(format("[%s]: Iceberg predicate expression: %s", signature, icebergExpression));
+
+    storeInUDFContext(ICEBERG_FILTER_EXPRESSION, icebergExpression);
+  }
+
+  private com.netflix.iceberg.expressions.Expression convert(Expression e) throws IOException {
+    OpType op = e.getOpType();
+
+    if (e instanceof BinaryExpression) {
+      Expression lhs = ((BinaryExpression) e).getLhs();
+      Expression rhs = ((BinaryExpression) e).getRhs();
+
+      switch (op) {
+        case OP_AND:
+          return and(convert(lhs), convert(rhs));
+        case OP_OR:
+          return or(convert(lhs), convert(rhs));
+        case OP_BETWEEN:
+          BetweenExpression between = (BetweenExpression) rhs;
+          return and(
+              convert(OP_GE, (Column) lhs, (Const) between.getLower()),
+              convert(OP_LE, (Column) lhs, (Const) between.getUpper())
+          );
+        case OP_IN:
+          return ((InExpression) rhs).getValues().stream()
+              .map((value) -> convert(OP_EQ, (Column) lhs, (Const) value))
+              .reduce(null, (m, v) -> (m == null ? v : or(m, v)));
```

Comment:
You can also use an initial value of `Expressions.alwaysFalse()` and remove the null check. `Expressions.or` will simplify `or(false, rhs)` to `rhs`.

Reply:
I think this is code style vs exactness tradeoff.  This implementation produces an expression tree that reflects the original expression whereas these other options produce one that is logically equivalent.  However, when you print the two (pig vs iceberg) you will see a difference that will look strange for an observer.

Reply:
Wait, I see what you're saying.  Looking at `Expressions.or` it immediately resolves that.  I'll update.